features/src/main/scala/com/salesforce/op/features/types/Text.scala

```diff
@@ -181,11 +182,11 @@ class URL(value: Option[String]) extends Text(value){
   /**
    * Extracts url domain, i.e. 'salesforce.com', 'data.com' etc.
    */
-  def domain: Option[String] = value map (new java.net.URL(_).getHost)
+  def domain: Option[String] = value map (s => new java.net.URL(new URI(s, false).toString).getHost)
   /**
    * Extracts url protocol, i.e. http, https, ftp etc.
    */
-  def protocol: Option[String] = value map (new java.net.URL(_).getProtocol)
+  def protocol: Option[String] = value map (s => new java.net.URL(new URI(s, false).toString).getProtocol)
```

Comment:
why do you need to escape the url when extracting protocol?

Reply:
cuz the way how we extracting protocol here always throw exception if any part of input contains special character

Reply:
Oh, got it.

Reply:
correct myself, it happens when host part of URL contains special character, i fixed this in test case